sonar-scm-svn-plugin/src/main/java/org/sonar/plugins/scm/svn/SvnScmProvider.java

```diff
@@ -99,21 +99,27 @@ public BlameCommand blameCommand() {
     return null;
   }
 
-  private static Set<Path> computeChangedPaths(Path rootBaseDir, SVNClientManager clientManager) throws SVNException {
+  private static Set<Path> computeChangedPaths(Path projectBasedir, SVNClientManager clientManager) throws SVNException {
     SVNWCClient wcClient = clientManager.getWCClient();
-    SVNInfo svnInfo = wcClient.doInfo(rootBaseDir.toFile(), null);
-    String base = "/" + Paths.get(svnInfo.getRepositoryRootURL().getPath()).relativize(Paths.get(svnInfo.getURL().getPath()));
+    SVNInfo svnInfo = wcClient.doInfo(projectBasedir.toFile(), null);
+
+    // SVN path of the repo root, for example: /C:/Users/JANOSG~1/AppData/Local/Temp/x/y
+    String svnRootPath = svnInfo.getRepositoryRootURL().getPath();
+    // SVN path of projectBasedir, for example: /C:/Users/JANOSG~1/AppData/Local/Temp/x/y/branches/b1
+    String svnProjectPath = svnInfo.getURL().getPath();
+    // path of projectBasedir, as "absolute path within the SVN repo", for example: /branches/b1
+    Path inRepoProjectPath = Paths.get(svnProjectPath.substring(svnRootPath.length()));
```

Comment:
Then here you could use relativize instead of substring.

Reply:
The problem of building a path directly from a URL path is that it may fail because of URL encoded characters. See https://community.oracle.com/blogs/kohsuke/2007/04/25/how-convert-javaneturl-javaiofile.
Maybe convert SVNURL to Path using:

    if ("file".equals(repositoryRootURL.getProtocol())) {
      Paths.get(new URL("file", repositoryRootURL.getHost(), repositoryRootURL.getPath()).toURI());
    }

and fail if protocol is not "file".